supported_os_controller.php

```diff
@@ -52,5 +56,217 @@ public function os()
+
+            // Loop through each serial number for processing
+            $out = array();
+            foreach ($machine->query($sql) as $serialobj) {
+                $out[] = $serialobj->serial_number;
+            }
+
+            $obj->view('json', array('msg' => $out));
+        } else {
+
+            $data = [];
+            $data["serial_number"] = $incoming_serial;
+            $data["reprocess"] = true;
+
+            // Process the serial in the model
+            $machine = new Supported_os_model();
+
+            // Send result
+            $obj->view('json', array('msg' => $machine->process($data)));
+        }
+    }
     
-} // END class Supported_os_module
+    /**
+     * Reprocess serial number
+     *
+     * @return void
+     * @author tuxudo
+     **/
+    public function recheck_highest_os($serial)
+    {
+        // Authenticate
+        if (! $this->authorized()) {
+            die('Authenticate first.'); // Todo: return json?
+        }
+        
+        $data = [];
+        $data["serial_number"] = $serial;
+        $data["reprocess"] = true;
+
+        // Process the serial in the model
+        $machine = new Supported_os_model();
+        $machine->process($data);
+        
+        // Send people back to the client tab once serial is reprocessed
+        redirect("clients/detail/$serial#tab_supported_os-tab");
+    }
+
+    /**
+     * Return JSON with information for admin page
+     *
+     * @return void
+     * @author tuxudo
+     **/
+    public function get_admin_data()
+    {
+        $obj = new View();
+        if (! $this->authorized()) {
+            $obj->view('json', array('msg' => 'Not authorized'));
+            return;
+        }
+
+        $current_os = munkireport\models\Cache::select('value')
+                        ->where('module', 'supported_os')
+                        ->where('property', 'current_os')
+                        ->value('value');
+        $source = munkireport\models\Cache::select('value')
+                        ->where('module', 'supported_os')
+                        ->where('property', 'source')
+                        ->value('value');
+        $last_update = munkireport\models\Cache::select('value')
+                        ->where('module', 'supported_os')
+                        ->where('property', 'last_update')
+                        ->value('value');
+
+        $obj = new View();
+        $out = array('current_os' => $current_os,'source' => $source,'last_update' => $last_update);
+        $obj->view('json', array('msg' => $out));
+    }
+
+    /**
+     * Retrieve data in json format
+     *
+     **/
+    public function get_data($serial_number = '')
+    {
+        $obj = new View();
+
+        if (! $this->authorized()) {
+            $obj->view('json', array('msg' => 'Not authorized'));
+            return;
+        }
+
+        $os = new Supported_os_model($serial_number);
+        $obj->view('json', array('msg' => $os->rs));
```

Comment:
There's now jsonView() that makes this more simple: `jsonView($os->rs);`

Reply:
Changed